src/sidebar/components/sidebar-content.js

```diff
@@ -234,27 +122,32 @@ function SidebarContentController(
   // Re-fetch annotations when focused group, logged-in user or connected frames
   // change.
   $scope.$watch(
-    () => [groups.focused(), store.profile().userid, ...store.searchUris()],
-    ([currentGroup], [prevGroup]) => {
-      if (!currentGroup) {
-        // When switching accounts, groups are cleared and so the focused group
+    () => [
+      store.focusedGroupId(),
+      store.profile().userid,
+      ...store.searchUris(),
+    ],
+    ([currentGroupId], [prevGroupId]) => {
+      if (!currentGroupId) {
+        // When switching accounts, groups are cleared and so the focused group id
         // will be null for a brief period of time.
         store.clearSelectedAnnotations();
         return;
       }
 
-      if (!prevGroup || currentGroup.id !== prevGroup.id) {
+      if (!prevGroupId || currentGroupId !== prevGroupId) {
         // The focused group may be changed during loading annotations as a result
         // of switching to the group containing a direct-linked annotation.
         //
         // In that case, we don't want to trigger reloading annotations again.
-        if (this.isLoading()) {
+        if (store.isFetchingAnnotations()) {
```

Comment:
This isn't checking for the document url being loaded now but I think that's ok here. If no document urls are loaded it will clear the annotations which should be empty anyway and even if it's not it will be emptied anyway once the urls are loaded. It will then attempt to load annotations but since no urls are present it will simply exit early.

Reply:
This block of code is also relevant when the group changes before annotations are loaded from the previous group I believe but I think in that case we would still want to load annotations so yes I think you are correct and this whole block can go now.